woocommerce/woocommerce-ios
# [Mobile Payments] Show onboarding when required during card present refunds

Part of: #6754


### Description


Interac refunds in Canada are processed via the card reader. There may also be other future cases in other countries, where refunds must also be done with the customer's card present.

These refunds initiate a connection to the card reader, so similar to the payments flows, we will check the merchant's onboarding status before we start connecting to a reader.

Related:
- #6761 
- #6770 
- #6774 

#### Note on testing

This commit removes the check for a card-present eligible payment gateway account which used to be part of the decision whether to show the `Collect Payment` button on Order Details or not. I did this to make the changes easier to test, and to be closer in line with Android and expected direction of the changes to the `Collect Payment` button.

To restore the previous behaviour, we will revert 1977780009ad413b526067278739a1d5e9f93839 before merging.

### Testing instructions


Using a Canada WCPay store:

1. Create an order, and take payment using an Interac card.
2. In `Menu > ⚙️ > In-Person Payments > Manage card reader`, disconnect the reader.
3. Open the Order Details, tap `Issue Refund`, and get through to the Refund Confirmation Screen
4. In your browser, deactivate the WCPay plugin
5. Tap `Refund` and confirm the alert
6. Observe that you see the "WCPay installed but not activated" onboarding screen
7. Activate the WCPay plugin using your browser
8. Tap `Refresh After Updating`
9. Observe that the onboarding flow is dismissed and the reader connection flow starts.
10. Connect a reader and follow the instructions to receive the refund.

Repeat the above steps with a non-Interac card, observe that the onboarding flow is not shown.
N.B. the refund will fail at the API call, because the plugin is disabled. But the onboarding flow is not used before that failure.

#### Behaviour with two plugins activated
I tried this with both Stripe and WCPay installed and activated too, as per previous PR testing steps.

In that case, the "Stripe not supported for IPP in Canada" onboarding step is shown, where I think we should prioritise the "Multiple Conflicting Plugins" in this case.

Also the "Stripe not supported" screen does not have a refresh control. I understand that design as it's sort of a dead-end failure case, but it could still be an improvement to the UX here to add pull to refresh controls to the various onboarding steps which don't yet have a CTA. This would remove the need to return to the Refund Confirmation screen and tap `Issue Refund` again, which is otherwise the only way to refresh that state.

I'll log issues for discussion of the above.

### Screenshots



https://user-images.githubusercontent.com/2472348/167150533-4715afd1-a411-4efe-b4c8-854574233709.mp4


---
